### PR TITLE
chore(eslint-plugin-internal): noUnnecessaryNoFormat

### LINKED ISSUE #6128: Repo: Flag instances of noFormat that don't do anything
### Suggestion

Some test suites, such as https://github.com/typescript-eslint/typescript-eslint/blob/027ce98b8af1a901ee4362cc252295bc03e6ca31/packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts, use `noFormat` because they need some code blocks to preserve formatting. But many of those `noFormat`s end up being copy & pasted across multiple tests, some of which end up not needing it.

Proposal: let's add a lint rule that flags uses of `noFormat` that don't need to be used?

:x: Incorrect:

```ts
noFormat`a.b;`
```

✅ Correct:

```ts
`a.b;`
```

### PR DESCRIPTION
**Pull request by Jolg42**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6128
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

I followed the proposal in https://github.com/typescript-eslint/typescript-eslint/issues/6128 and added a check for unnecessary `NoFormat` occurrences, with a test.

### CODE REVIEW

File: packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts [hunk 1]
```diff
@@ -157,6 +160,7 @@ export default createRule<Options, MessageIds>({
     function prettierFormat(
       code: string,
       location: TSESTree.Node,
+      silent?: boolean,
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts:163]: [Refactor] Hmm, I'm not generally a fan of adding boolean parameters to modify core behavior of a function. It makes it hard to keep track of the function's responsibilities & adds to its complexity. I think the code would be easier to wrangle if we extract out the logic we actually care about from the existing function into a new one.

How about this  of function signatures?

* `function getCodeFormatted(code: string): string | FormattingError | null`
* `function getCodeFormattedOrReport(code: string, location: TSESTree.Node): string | null`

...where `FormattingError` indicates the error message, `getCodeFormatted` contains the call to Prettier, and `getCodeFormattedOrReport` contains the call to `context.report`?

File: packages/eslint-plugin-internal/tests/rules/plugin-test-formatting.test.ts [hunk 2]
```diff
@@ -128,7 +127,7 @@ const a = 1;
       ],
     },
 
-    // empty linems are valid when everything else is indented
+    // empty lines are valid when everything else is indented
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin-internal/tests/rules/plugin-test-formatting.test.ts:130]: [Praise] 😄 nice spot

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Functionality looks great, thanks for sending this @Jolg42! Requesting changes on the internals, but if you hate what I'm suggested we can certainly ignore me. 😄

**Comment by JoshuaKGoldberg** [comment]
👋 @Jolg42 just checking in, did you want to re-request our review on this one? And/or, is this still something you have time to work on?

No worries if not, I just don't want to leave it dangling. And it is a very lovely change 😄

**Comment by Jolg42** [comment]
@JoshuaKGoldberg Thanks for checking-in and I'm sorry for not answering earlier, I still have to work on this 😅

**Comment by JoshuaKGoldberg** [comment]
No worries! Next time I'm in Berlin, we can always pair on it. 😄

**Comment by JoshuaKGoldberg** [comment]
Closing this PR as it's been stale for a few months without activity. Feel free to reopen @Jolg42 if you have time - but no worries if not! If anybody wants to drive it forward, please do post your own PR - and if you use this as a start, consider adding `Co-authored-by: @Jolg42` at the end of your PR description. Thanks! 😊

**Pull request closed without merging**